### PR TITLE
Upgrade vaadin and related versions for Dev UI

### PR DESCRIPTION
**Pull request by phillip-kruger**

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

**Review by gsmet — CHANGES_REQUESTED** [review]
I'd like us to not rush this as I'm not sure it actually solves the issue.

Your rationale for not enforcing the versions was that you wanted extensions like web bundler to be able to use other versions... and in the end we add a new BOM to actually enforce the versions.

I'm not entirely sure it brings us anything in the end.

Could you explain how having a separate BOM actually brings us anything compared to the previous situation of having things in the Quarkus BOM?

**Comment by phillip-kruger** [comment]
The quarkus bom is used by the user's application, the new bom is only used in die extension deployment module. So now a user can add UI library without Dev UI influence (with or without the web bundler)

**Comment by maxandersen** [comment]
Unless we want to keep "leaking" devmode only relevant dependencies into all extensions and user apps something gotta give and I think this solution of having a dev-ui bom is the most pragmatic/proper solution - even though it does require many extensions to get updated to have decent build reproducability.

I've asked @phillip-kruger to make a draft PR in a quarkiverse extension to show the exact impact.

Also asked to check how far back this actually impacts us - the assumption is before September quarkus-core was locked by bom and parent-bom but any non-quarkus ext was impacted unless they use quarkus core bom directly. 

And I suggest we write up the alternates in an ADR and make a recorded decision on how we deal with this because we do need those dependencies locked between releases.

File: docs/src/main/asciidoc/writing-extensions.adoc [hunk 1]
```diff
@@ -1596,6 +1596,23 @@ Instead, Quarkus extensions can make use of various xref:cdi-reference.adoc[Buil
 
 You can make your extension support the xref:dev-ui.adoc[Quarkus Dev UI] for a greater developer experience.
 
+[IMPORTANT]
+====
+If your extension is NOT part of Quarkus core, you can add this BOM in the dependency management of your extension parent pom to lock the javascript versions:
```
Comment (maxandersen) [inline docs/src/main/asciidoc/writing-extensions.adoc:1601]:
```suggestion
If your extension is NOT part of Quarkus core, you should add this BOM in the dependency management of your extension parent pom to lock the javascript versions:
```

**Comment by metacosm** [comment]
So, if I understand correctly, extension developers that wish to leverage components provided by the Dev UI will now need to import this new BOM, right?
If an extension doesn't provide any contribution to the Dev UI, this new BOM can be ignored, right?

**Comment by phillip-kruger** [comment]
@metacosm no, it always have to be there if the extension use vertx-http. But rather than adding another bom, we might just create a new bom (we are still discussing). I am closing this PR for now, we are rolling back the bom move, and then I'll play with some options.

**Pull request closed without merging**